numpy/core/src/multiarray/alloc.c

```diff
@@ -433,11 +433,26 @@ int uo_index=0;   /* user_override index */
 
 /* Wrappers for the default or any user-assigned PyDataMem_Handler */
 
+int default_allocator_policy = 1;
+
+static inline PyDataMem_Handler *
+_PyDataMem_GetHandler_Internal(PyObject *mem_handler)
+{
+#if (!defined(PYPY_VERSION_NUM) || PYPY_VERSION_NUM >= 0x07030600)
+    if (default_allocator_policy)
```

Comment:
Would it make sense to do `PyDataMem_DefaultHandler == mem_handler` here instead?  That way it will remain working if the default had been changed once.

Otherwise, I am OK with this (modulo possible style nitpicks), so long @mattip doesn't mind the added complexity.

Reply:
@mattip I rebased to main. Could you review?